### ISSUE TITLE
Support for Point Clouds

### ISSUE THREAD

**Issue by daniel-j-h** [mapbox/mapbox-gl-js#3450]
I'm currently working on a library for a LiDAR and wanted to make use of Mapbox GL JS for visualizing point clouds both in real-time (while driving your car) but also statically (for the route).

For this use-case here are the road-blockers I encountered.

1/ I'm not able to extrude points. I can hack around this with small polygons. But then it seems like this is roughly the smallest polygon I can get to show up (there's probably a filter for noise?):

![Smallest extruded polygons](https://cloud.githubusercontent.com/assets/527241/19661163/7d4c9a6e-99e7-11e6-971d-a14ef6fb50f8.png)

2/ The extrusion feature seems to not support values < 1.
Here's what happens for a fractional 0.5 extrusion value:

![Fractional extrusion height](https://cloud.githubusercontent.com/assets/527241/19661162/7d4a025e-99e7-11e6-9546-97b5259b5103.png)

3/ There is no streaming source. My plan was to either use a Websocket server pushing point clouds into the browser and updating a source or exposing a GeoJSON endpoint which changes every second.

I don't expect support for point clouds in the foreseeable future.
I just want to bring blockers for the point cloud / LiDAR use-case to your attention.

**Comment by lbud** [comment]
@daniel-j-h right, extruded polygons aren't the right avenue for this. Point cloud support will become possible once we add z offset properties to different layer types (https://github.com/mapbox/mapbox-gl/issues/6#issuecomment-244193192) — z-offset circle types would seem to be the most logical display here, a la
![image](https://cloud.githubusercontent.com/assets/3170312/19669818/3d0814bc-9a17-11e6-9d90-21505c8d103e.png)

**Comment by mariokam** [comment]
Hi there! z-offset is exactly what I need also! Any info when could be expected? Thanks!

**Comment by lbud** [comment]
@mariokam z-offset properties are on our long-term roadmap, but aren't prioritized in the near term (Q3).

**Comment by Scarysize** [comment]
@mariokam Did you take a look at deck.gl? It works on top of mapbox and has a point cloud layer: https://uber.github.io/deck.gl/#/documentation/layer-catalog/point-cloud-layer

**Comment by robinbezak** [comment]
@Scarysize Hey there! We need to extrude symbol custom marker :D like u can extrude polygon a give to it some height and base_height the same I need for custom marker :D Iam using more floor building extrude but i want markers on every floor :D I dont know what to do :(

**Comment by mariokam** [comment]
any updates on this topic? I think that if the polygon extrusion is working, there could be some beta version released already on points as well.

**Comment by andrewharvey** [comment]
Linking to https://github.com/mapbox/vector-tile-spec/issues/35 which could hopefully make encoding point clouds in Mapbox Vector Tiles more efficient.

**Comment by mariokam** [comment]
Is it still open? Are there any plans for extruding points in the roadmap?

**Comment by andrewharvey** [comment]
@mariokam Take a look at https://blog.mapbox.com/hd-vector-maps-open-standard-335a49a45210

**Comment by exotfboy** [comment]
@andrewharvey 👍 

I am interested in how is this implemented?  Sounds like the vector tile is using the vector tile  specification 3, but when I dig into the example, I found it use the mapbox-gl-js 0.45.0-beta. So how can it support the point to be rendered with the height attribution?

**Comment by andrewharvey** [comment]
I think the example at https://blog.mapbox.com/hd-vector-maps-open-standard-335a49a45210 is using a branch of mapbox-gl-js that adds support for circles with a z coordinate.

**Comment by bravecow** [comment]
@exotfboy I tied. Didn't work:

https://bravecow.github.io/mapbox-3d/#16.7/44.058619/-123.06807/-170.9

Some guy here got the same result: points not rendered in Z index.

**Comment by xyz-mtl** [comment]
Do we have a status about this request?
 Point cloud is a critical feature for the future of web mapping application. I don’t understand why this feature is not supported by Mapbox ....

**Comment by patrickkeenan** [comment]
Any update on this?

**Comment by exotfboy** [comment]
> Any update on this?

Maybe you can check ESRI Arcgis Javascript API, it support PointCloud and other 3D like layers, however I am not sure the data format of the PointCloud Layer.

**Comment by ondrejrohon** [comment]
Any updates on this in 2021? :)

**Comment by Khawlamira** [comment]
Any updates in 2022 ?

**Comment by mjnks** [comment]
Pre-emptive plug for 2023.

**Comment by aaronw-dev** [comment]
Nothing, huh?
